barretenberg/cpp/src/barretenberg/dsl/acir_format/honk_recursion_constraint.cpp

```diff
@@ -139,13 +130,16 @@ AggregationObjectIndices create_honk_recursion_constraints(Builder& builder,
         builder.assert_equal(builder.add_variable(1 << log_circuit_size), key_fields[0].witness_index);
         builder.assert_equal(builder.add_variable(input.public_inputs.size()), key_fields[1].witness_index);
         builder.assert_equal(builder.add_variable(UltraFlavor::has_zero_row ? 1 : 0), key_fields[2].witness_index);
-        builder.assert_equal(builder.add_variable(0), key_fields[4].witness_index);
+        builder.assert_equal(builder.add_variable(1), key_fields[4].witness_index);
         uint32_t offset = 4;
-        // TODO(https://github.com/AztecProtocol/barretenberg/issues/1059): Properly set this to a default agg obj.
+        size_t num_inner_public_inputs = input.public_inputs.size() - bb::AGGREGATION_OBJECT_SIZE;
+
+        // We are making the assumption that the aggregation object are behind all the inner public inputs
```

Comment:
not sure what the right thing to do here is. Maybe it doesn't matter. I think Plonk sets all of these to be 0